### PR TITLE
Split dotnet/runtime-only downlevel LibraryImport support into its own source generator

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Instead of shipping unusable downlevel support in LibraryImportGenerator and having to maintain how it works in combination with the non-downlevel support that we actually expect users to use, split the generator into two separate generators:

- LibraryImportGenerator: Our shipping current-TFM support
- DownlevelLibraryImportGenerator: dotnet/runtime-only .NET Standard + .NET Framework support.

The Downlevel support only needs to support standard and NETFX as all .NETCoreApp TFMs we target for downlevel support all ship LibraryImportGenerator already (and they build using the version that shipped with the TFM).

This PR also shares the managed->unmanaged stub generation between LibraryImportGenerator and ComInterfaceGenerator to reuse more code cleanly.

The MSBuild changes in Directory.Build.targets help fix a problem where the TFM-included generators would be removed in the generator unit test projects, which also reference the projects directly as regular (non-analyzer) references. This would fail sporadically when working locally before this change, but the changes in generators.targets caused the intermittent failure to become consistent.

### CODE REVIEW

**Review by jtschuster — APPROVED** [review]
Don't see any issues aside from the JS generator test failures.

File: src/libraries/System.Runtime.InteropServices/gen/DownlevelLibraryImportGenerator/LanguageSupport.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply src/libraries/System.Runtime.InteropServices/gen/DownlevelLibraryImportGenerator/LanguageSupport.cs:1]: We'd end up having both generators referenced at the same time. I can share more of these files in a Common-style folder though.

**Comment by jkoritzinsky** [comment]
/ba-g failure is a network timeout that looks generally flaky

File: eng/generators.targets [hunk 2]
```diff
@@ -1,51 +1,56 @@
...
+                                  '$(IsSourceProject)' == 'true' and
+                                  '$(MSBuildProjectExtension)' == '.csproj' and
                                   (
-                                    !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0')) or
-                                    (
                                       '$(DisableImplicitFrameworkReferences)' == 'true' and
                                       (
                                         '@(Reference->AnyHaveMetadataValue('Identity', 'System.Runtime.InteropServices'))' == 'true' or
-                                        '@(ProjectReference->AnyHaveMetadataValue('Identity', '$(CoreLibProject)'))' == 'true'
+                                        '@(Reference->AnyHaveMetadataValue('Identity', 'System.Private.CoreLib'))' == 'true'
```
Comment (ViktorHofer) [inline eng/generators.targets:36]: This change looks wrong. System.Private.CoreLib is never a reference as it isn't part of the targeting pack.

**Review by ViktorHofer — COMMENTED** [review]
This change requires updating all the solution files as this is adding a new dependency. Can you please do that? The command is `.\dotnet.cmd build src\libraries\slngen.proj`.

Please also check if the System.Private.CoreLib slns need to be updated.

File: eng/generators.targets [hunk 3]
```diff
@@ -1,51 +1,56 @@
...
+         - doesn't target the a TFM that includes LibraryImportGenerator -->
+    <EnabledGenerators Include="DownlevelLibraryImportGenerator"
                        Condition="'$(EnableLibraryImportGenerator)' == '' and
                                   (
                                     '$(IsSourceProject)' == 'true' or
                                     '$(IsTestProject)' == 'true' or
                                     '$(IsTestSupportProject)' == 'true'
                                   ) and
                                   '$(MSBuildProjectExtension)' == '.csproj' and
+                                  !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppMinimum)'))" />
```
Comment (ViktorHofer) [inline eng/generators.targets:22]: This should just be `$(TargetFrameworkIdentifier)' != '.NETCoreApp'` as we shouldn't  target an out-of-support .NETCoreApp TFM.

File: eng/generators.targets [hunk 4]
```diff
@@ -1,51 +1,56 @@
 <Project>
 
   <PropertyGroup>
+    <!-- Enable LibraryImportGenerator for CoreLib. -->
```
Comment (ViktorHofer) [inline eng/generators.targets:4]: I understand that this already existed before this change but it feels weird to special case CoreLib in here. Can we move this property into the shared CoreLib project instead?

**Comment by jkoritzinsky** [comment]
@ViktorHofer I've addressed your feedback in https://github.com/dotnet/runtime/pull/106827

**Pull request merged**